strings/base_meta.h

```diff
@@ -159,7 +159,7 @@ namespace winrt::impl
     {
         operator I() const noexcept
         {
-            return static_cast<D const*>(this)->template try_as<I>();
+            return static_cast<D const*>(this)->template as_or_failfast<I>();
```

Comment:
This feels like a pretty big change. I think this means code like `auto stringable = static_cast<IStringable>(lies)` will crash? That's a legit way of probing the type hierarchy, like this:

```c++
if (auto stringable = static_cast<IStringable>(lies)) {
    stringable.ToString();
}
```

... but yes it's strange that `static_cast<>` here can "fail."

See WINRT_IMPL_SHIM - that's the place that should get this fix.  If this was limited to the cast-shim helper on the projected runtimeclass type it might be better, since it's no "worse" than we have today (and in fact better since it buckets more nicely.)

Note @oldnewthing 's comment though - this means all the `foo.Bar(); foo.Bas()` that use the shim now get additional code generated to check the hresult and throw/failfast.

I suggest rebuilding something meaty internally with this change in place (Baker, etc.) and see what the codegen looks like using sizebench.